# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundParentImpl.cpp
# Commit: f5921ffeaee4
# Full Hash: f5921ffeaee4b8b096f9999d4171fee1c3e1c57a
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-08-05 16:34:46
# Description:
#   Bug 1723954: Return NS_OK if PrincipalInfoToPrincipal fails. r=ckerschb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D121867
# ==============================================================================

diff -r 5e85a00a7b30 -r f5921ffeaee4 ipc/glue/BackgroundParentImpl.cpp
--- a/ipc/glue/BackgroundParentImpl.cpp	Thu Aug 05 13:05:39 2021 +0000
+++ b/ipc/glue/BackgroundParentImpl.cpp	Thu Aug 05 13:58:41 2021 +0000
@@ -896,6 +896,7 @@
     if (NS_WARN_IF(principalOrErr.isErr())) {
       mContentParent->KillHard(
           "BroadcastChannel killed: PrincipalInfoToPrincipal failed.");
+      return NS_OK;
     }
 
     nsAutoCString origin;
